src/vellum/workflows/nodes/core/map_node/node.py

```diff
@@ -173,6 +173,7 @@ def _run_subworkflow(self, *, item: MapNodeItemType, index: int) -> None:
         subworkflow = self.subworkflow(
             parent_state=self.state,
             context=context,
+            store=EmptyStore(),
```

Comment:
woah! this is definitely a huge source of the memory leak. let's make the following updates:
- Let's update the `__init__` of `WorkflowContext` to accept an optional prop for the `Type[Store]`
- Let's update the `__init__` of `BaseWorkflow` to always prioritize the type of store assigned to WorkflowContext before defaulting. Add a comment here that in v2.0.0, we should remove the concept of an internal store altogether.
- Let's update the `WorkflowContext.create_from` to pass down the store type